### ISSUE TITLE
[Feature] [QoL] Add the option to cache the thumbnail

### ISSUE THREAD

**Issue by Kazer67** [FuzzyGrim/Yamtrack#241]
Instead of loading the thumbnail for the collection from various third-party source, it would be nice to have an option (maybe either per users in the settings or directly servers wide?) to cache them locally when added.

The purpose would give an option, especially for those who host Yamtrack at home, to load directly from the LAN instead of the WAN, if they want (which also would make thumbnail load in the event of a downtime from the third-party website).

For example with the Manga category (but should apply to the rest), thumbnail (chosen randomly for this example) are between 100 to 200KB  which can be considered very low especially for those with high bandwidth but this need to be multiplied by the number of Manga in the list which may increase (in my case, with around 42 mangas the network used with external thumbnail is 6,72MB while it only use 736,87KB when I'm blocking the loading of images from third-party).

Maybe using a dedicated folder?

**Comment by FuzzyGrim** [comment]
In the early stages of the application I did store the images locally, but Django, the framework I use to develop the application, needs something like Nginx to serve the local images when running in production, which made the self-hosting setup a bit more complex.

I will have to think about it, and investigate further if there are other options.

**Comment by skibbipl** [comment]
Yeah, I would also like to have such functionality. I've switched from mediatracker and it caches the image data locally.

**Comment by BraianPita** [comment]
I don't think an external service should be needed, I was able to find this documentation:
https://docs.djangoproject.com/en/5.2/howto/static-files/

**Comment by BraianPita** [comment]
I would really like this feature to be added. As far as I can see, the app already is running an nginx  instance as part of the deployment with an nginx.conf file which has a static files folder, I don't think it would complicate the deployment setup?

**Comment by christaikobo** [comment]
This one gets my vote too. I also have a related question, for the time being, are information of an entry, like synopsis, tags, release date, chapters, platforms, etc. stored locally? In the case of temporary API disruption or just limit on API usage, will the entries I marked be available for view locally?

**Comment by FuzzyGrim** [comment]
API requests are cached, to prevent rate limits and short API downtimes.

**Comment by christaikobo** [comment]
Thanks for the answer, how long is it cached for? Where is the cache directory?

EDIT: I found out, so currently it is cached for `CACHE_TIMEOUT` in redis. 

I think most people would prefer caching it on disk as opposed to only in redis? I opened https://github.com/FuzzyGrim/Yamtrack/issues/652 for it.

**Comment by fdw** [comment]
Would love this feature as well. Maybe it would be possible to use Jellyfin's posters, to save on space and to use the same ones?

**Comment by hreczkosiej** [comment]
Would be great to have this feature

**Comment by alogieishara** [comment]
I'd love this feature for long term preservation of my database, it's scary to depend on a bunch of APIs and links.

**Comment by christaikobo** [comment]
For long term data preservation you should vote on https://github.com/FuzzyGrim/Yamtrack/issues/652 instead of this issue, this issue is about caching, and even if thumbnail is cached it will be expired in 24 hours.

**Comment by GlassedSilver** [comment]
> For long term data preservation you should vote on [#652](https://github.com/FuzzyGrim/Yamtrack/issues/652) instead of this issue, this issue is about caching, and even if thumbnail is cached it will be expired in 24 hours.

Well, technically you need to vote for both then, because if the cache is written to disk, but no images are loaded to the cache but only referenced externally in-browser then nothing will be stored on disk.

That being said I'm going to vote for both, seems like the two aspects are the essence of why people want to self-host:
Sovereignty over their data and its usage
